### ISSUE TITLE
z plugin tab completion doesn't work

### ISSUE THREAD

**Issue by ricpelo** [ohmyzsh/ohmyzsh#10975]
### Describe the bug

Using Debian testing (bookworm), fully updated as of today.

Using z plugin with Oh-My-Zsh doesn't work. When I type `z` and then press the `Tab` key, instead of shows the most frequent directories, it shows the following output:

![Captura de pantalla_2022-06-04_15-54-52](https://user-images.githubusercontent.com/616169/172003946-237cbe24-b4fe-4b33-a807-0b3a8f91fef5.png)

However, when I use the `z` plugin without Oh-My-Zsh (i.e. only using Zsh and then `source ~/.oh-my-zsh/plugins/z/z.zsh`, it works OK:

![Captura de pantalla_2022-06-04_15-59-51](https://user-images.githubusercontent.com/616169/172004050-6864069f-7e0c-45ce-86c8-898d494bd53a.png)

### Steps to reproduce

1. Enable plugin `z`.
2. Type `z` + press `Space` + press `Tab` key.

### Expected behavior

It shows the most frequent directories and I can select one using the `Tab` key.

### OS / Linux distribution

Debian testing (bookworm)

### Zsh version

zsh 5.9 (x86_64-debian-linux-gnu)

### Terminal emulator

Alacritty

**Comment by rwmitchell** [comment]
At least in my case, the incorrect completions are coming from:

```local/share/zsh/5.8.1.1-dev/functions/_compadd```

which suggests the issue isn't with OMZ directly, but OMZ's use of compadd that triggers the problem.

**Comment by mcornella** [comment]
OMZ does not use `compadd` to change the z completion. It doesn't do anything extra apart from calling `compinit`. When that is done in a non-OMZ shell with zsh 5.9, the bug appears again. 

It looks like there's an incompatibility with how z uses compctl instead of compsys and zsh 5.9. I'm still investigating.

**Comment by rodhash** [comment]
**oh-my-zsh** uses rupa's Z, right?


Not sure whether or not rupa is willing to make it compatible with newer ZSH versions, however, this gentlemen made it compatible:
https://github.com/agkozak/zsh-z

I'm using it for a little bit and seems to be working fine with zsh 5.9.

Given rupa isn't a fan of ZSH (see issue 100 if curious) perhaps this version above is the best choice atm.

**Comment by carlosala** [comment]
It looks really interesting, it's probably an option to change to that!

**Comment by ricpelo** [comment]
I've just tried the @rodhash' solution and it works very well! Like @carlosala said, to me it could be great if `rupa/z` would be replaced by `agkozak/zsh-z`.

**Comment by rodhash** [comment]
Yeah, also cool that he reports to have native ZSH support, so, sounds a good deal to me .. if you wish I'd be happy to contribute w/ a PR ..

I just need to fix my PC, I messed up my grub so I'm dual booting to fix it

**Comment by carlosala** [comment]
Oh grub issues... Good luck with that!

**Comment by rodhash** [comment]
Yeah, I was experimenting with a new distro and something weird messed my grub in the process .. now it's good again.

So you guys think it's a good idea to replace **z** completely or just add the **zsh-z** (keeping z for whatever reason)? I haven't yet read the contributing.md, will check that out.

In my local repo I just added zsh-z and seems to be working fine (being called by oh-my-zsh and not the custom one).

**Comment by ricpelo** [comment]
@rodhash Both options looks good to me, but I prefer replacing `z` completely, as one of its main functionality simply doesn't work and it's a tested and working alternative.

**Comment by rodhash** [comment]
Hello, sir!

Awesome job! I'm happy **oh-my-zsh** folks accepted your plugin as a replacement for z :)

> I'm glad my [Zsh-z](https://github.com/agkozak/zsh-z) plugin solves your problem. Let me know if you have any questions about it.
